Capabilities

alessandropellegrini opened the pull request:
Capabilities subsystem

alessandropellegrini committed:
Configuration querying function

We introduce the `CapabilityAvailable()` runtime API, which allows
models to know what capabilites are available (at compile time or by
enabling/disabling them using command line switches).

This is beneficial to simulation models, so that we can setup different
execution paths depeding on the available subsystems, and will
eventually allow to make all different versions of the models converge
in a single reference implementation, thus increasing maintainability.

Signed-off-by: Alessandro Pellegrini <pellegrini@dis.uniroma1.it>

alessandropellegrini committed:
Introduce unit tests for capabilities

A simple unit test for the capabilities subsystem is introduced.

Signed-off-by: Alessandro Pellegrini <pellegrini@dis.uniroma1.it>

Piccions commented on src/ROOT-Sim.h:
```diff
@@ -220,3 +221,61 @@ void 			RemoveVisit		(agent_t agent, unsigned i);
+struct capability_info_t {
+	enum capability_t capability;
+	union {
+		char *output_dir;		///< Destination Directory of output files
```
I feel like this should be of type `const char *`

Piccions reviewed (dismissed):
Good PR, I appreciate that you wrote a test.

alessandropellegrini committed:
Slightly better doc of Capability API

This commit introduces a bit of more information in the documentation
for the Capability API.

Additionally, as suggested by Piccions, we are dumb-proofing the pointer
to the output configuration directory---of course, users could always
screw up everything in there if they really want, or ignore compiler
warning.

Signed-off-by: Alessandro Pellegrini <pellegrini@dis.uniroma1.it>

Piccions reviewed (approved):
Perfect PR, thanks Alessandro!
